core/src/main/scala/org/bitcoins/core/crypto/AesCrypt.scala

```diff
@@ -0,0 +1,165 @@
+package org.bitcoins.core.crypto
+
+import java.security.SecureRandom
+
+import javax.crypto.spec.{IvParameterSpec, SecretKeySpec}
+import javax.crypto.{BadPaddingException, Cipher, SecretKey}
+import scodec.bits.ByteVector
+
+import scala.util.{Failure, Success, Try}
+
+case class AesEncryptedData(
+    cipherText: ByteVector,
+    iv: ByteVector,
+    salt: String)
+
+/**
+  * Provides functionality for encrypting and decrypting with AES
+  *
+  * @note Most of the content here is taken from
+  *       [[https://stackoverflow.com/questions/992019/java-256-bit-aes-password-based-encryption this SO question]]
+  */
+object AesCrypt {
+
+  /**
+    * AES encryption with block cipher mode and PBKDF2
+    *
+    * @see [[https://stackoverflow.com/questions/1220751/how-to-choose-an-aes-encryption-mode-cbc-ecb-ctr-ocb-cfb SO question]]
+    *     on different cipher modes with AES
+    *
+    * @see [[https://www.ietf.org/rfc/rfc2898.txt IETF RFC2898: PKCS #5: Password-Based Cryptography Specification]]
+    *     on PBKDF2
+    */
+  private val aesCipherType: String = "AES/CBC/PKCS5Padding"
+
+  private def getCipher: Cipher = {
+    val cipher = Cipher.getInstance(aesCipherType)
+    cipher
+  }
+
+  private def decryptionCipher(
+      secret: SecretKey,
+      initializationVector: ByteVector): Cipher = {
+    val cipher = getCipher
+    cipher.init(Cipher.DECRYPT_MODE,
+                secret,
+                new IvParameterSpec(initializationVector.toArray))
+    cipher
+  }
+
+  /**
+    * Decrypts the provided data
+    */
+  def decrypt(
+      encrypted: AesEncryptedData,
+      password: String): Either[AesDecryptionException, ByteVector] = {
```

Comment:
We should really consider making a type for passwords that wraps a String so that we never accidentally pass a password anywhere it isn't expected.